Surface_sweep_2/include/CGAL/Surface_sweep_2/Surface_sweep_2_impl.h

```diff
@@ -520,9 +517,8 @@ template <typename Vis>
 
   CGAL_assertion(c1 != c2);
 
-  // look up for (c1,c2) in the table and insert if doesnt exist
-  Curve_pair cv_pair(c1,c2);
-  if (! (m_curves_pair_set.insert(cv_pair)).second) {
+  // look up for c1 in the table of c2 (or vice versa if c2<c1) and insert if doesnt exist
+  if ((c1 < c2 ? c1->intersection_exists(c2) : c2->intersection_exists(c1))) {
```

Comment:
Maybe it is worth calculating in advance some kind of middle address?
Otherwise, there is a skew in filling
```
bool insert;
if (c1 < middle && c2 < middle) {
    insert = c1 < c2 ? c1->intersection_exists(c2) : c2->intersection_exists(c1);
} else {
    insert = c1 < c2 ? c2->intersection_exists(c1) : c1->intersection_exists(c2);
}
```
Maybe
```
if (c1 < middle && c2 < middle) {
    insert = c1 < c2 ? c1->intersection_exists(c2) : c2->intersection_exists(c1);
} else if (c1 < middle) {
    insert = (middle - c1) < (c2 - middle) ? c1->intersection_exists(c2) : c2->intersection_exists(c1);
} else if (c2 < middle) {
    insert = (middle - c2) < (c1 - middle) ? c2->intersection_exists(c1) : c1->intersection_exists(c2);
} else {
    insert = c2 > c1 ? c2->intersection_exists(c1) : c1->intersection_exists(c2);
}
```
Another option is to check for availability in both and insert in the one with less. Then the size will be "uniform"

Reply:
I don't really understand what you mean by "middle" address (if it is really the "middle", as in `(c1+c2)/2`, then I don't see the point of testing `c1<middle && c2<middle` which will obviously return false everytime).

And overall I don't see the point of this suggested modification. Why would we want size to be "uniform"?

Reply:
I mean the global median, if you can count it

but it may be difficult, and it looks strange, I like the option with lookup in both maps more

Reply:
we may want a uniform size to increase the speed of the lookup, and to increase the chance that we will avoid allocation.  

For example, now the curve with the smallest address will collect all its intersects